src/engraving/draw/transform.h

```diff
@@ -20,27 +20,153 @@
+    };
+
     Transform() = default;
-    Transform(const QTransform& t)
-        : QTransform(t) {}
 
-    PointF map(const PointF& p) const { return PointF::fromQPointF(QTransform::map(p.toQPointF())); }
-    LineF map(const LineF& p) const { return LineF::fromQLineF(QTransform::map(p.toQLineF())); }
-    PainterPath map(const PainterPath& p) const { return QTransform::map(p); }
+    Transform(double h11, double h12, double h21, double h22, double dx, double dy);
+
+    Transform operator*(const Transform& m) const;
+    Transform& operator*=(const Transform& o);
+    Transform& operator*=(double num);
+    Transform& operator/=(double div);
+
+    double m11() const { return m_affine.m_11; }
+    double m12() const { return m_affine.m_12; }
+    double m13() const { return m_13; }
+    double m21() const { return m_affine.m_21; }
+    double m22() const { return m_affine.m_22; }
+    double m23() const { return m_23; }
+    double m31() const { return m_affine.m_dx; }
+    double m32() const { return m_affine.m_dy; }
+    double m33() const { return m_33; }
+    double dx() const { return m_affine.m_dx; }
+    double dy() const { return m_affine.m_dy; }
+
+    PointF map(const PointF& p) const;
+    LineF map(const LineF& l) const;
+    PainterPath map(const PainterPath& path) const;
+
+    const Matrix& toAffine() const { return m_affine; }
+
+    void setMatrix(double m11, double m12, double m13, double m21, double m22, double m23, double m31, double m32, double m33);
+
+    void reset();
+
+    Transform& rotate(double a);
+    Transform& rotateRadians(double a);
+    Transform& translate(double dx, double dy);
+    Transform& scale(double sx, double sy);
+    Transform& shear(double sh, double sv);
+    Transform inverted() const;
+
+#ifndef NO_QT_SUPPORT
+    static QTransform toQTransform(const Transform& transform);
+    static Transform fromQTransform(const QTransform& transform);
+#endif
+
+private:
+
+    Transform(double h11, double h12, double h13, double h21, double h22, double h23, double h31, double h32, double h33);
+
+    inline double determinant() const
+    {
+        return m_affine.m_11 * (m_33 * m_affine.m_22 - m_affine.m_dy * m_23)
+               - m_affine.m_21 * (m_33 * m_affine.m_12 - m_affine.m_dy * m_13) + m_affine.m_dx
+               * (m_23 * m_affine.m_12 - m_affine.m_22 * m_13);
+    }
+
+    Transform adjoint() const;
+
+    Transform::TransformationType type() const;
+
+    inline Transform::TransformationType inline_type() const
+    {
+        if (m_dirty == TransformationType::None) {
+            return m_type;
+        }
+        return type();
+    }
 
-    Transform inverted(bool* invertible = nullptr) const { return Transform(QTransform::inverted(invertible)); }
+    void mapElement(double& nx, double& ny, TransformationType t) const;
+
+#ifdef DEBUG
+    static void nanWarning(const std::string& func)
+    {
+        LOGW() << "Transform:: " << func << " with NaN called";
+    }
```

Comment:
It is better to move the implementation to `.cpp` because it is better not to use the log in the `.h` file because there are many different macros in the `log.h` file that can conflict with macros from standard libraries (especially in Windows), this can lead to bad consequences.
It is better to include `log.h` only in the `.cpp` file and after all the inclusions.